gadget/install/install.go

```diff
@@ -324,6 +359,31 @@ func onDiskStructsSortedIdx(vss map[int]*gadget.OnDiskStructure) []int {
 	return yamlIdxSl
 }
 
+func determineDeviceVolumes(gi *gadget.Info) (map[string]gadget.DeviceVolume, error) {
```

Comment:
Feels like the auxiliary DeviceVolume adds some of the complexity. I haven't played with this, but what if instead of adding external structure, we extend the Volume stcuture with a `RuntimeDevice` or something similar. Then once an assignment match has been established, there'd be a call eg. `vol.AdoptRuntimeDeviceAssignment(..)`, which updates RuntimeDevice accordingly. Then instead of dealing with DeviceVolume we keep using Volume like before.

Reply:
I like this suggestion